### ISSUE TITLE
Update dependencies

### ISSUE THREAD

**Issue by michielbdejong** [solid/mashlib#94]
## Step 1

Using the latest npm versions of:
* rdflib@1.2.3
* solid-auth-cli@1.0.14
* solid-auth-client@2.4.1
* solid-auth-tls@0.1.2

I'm updating npm dependencies for:
* [x] [solid-namespace](https://github.com/solid/solid-namespace/pull/14) ()
* [x] [pane-registry](https://github.com/solid/pane-registry/pull/7) (rdflib)
* [x] [solid-ui](https://github.com/solid/solid-ui/pull/310) (pane-registry, rdflib, solid-auth-client, solid-auth-tls, solid-namespace)
* [x] [chat-pane](https://github.com/solid/chat-pane/pull/39) (solid-ui)
* [x] [contacts-pane](https://github.com/solid/contacts-pane/pull/15) (solid-ui)
* [x] [folder-pane](https://github.com/solid/folder-pane/pull/13) (solid-ui)
* [x] issue-pane (solid-ui)
* [x] meeting-pane (solid-ui)
* [x] source-pane (solid-ui)
* [x] [solid-panes](https://github.com/solid/solid-panes/pull/231) (chat-pane, contacts-pane, folder-pane, issue-pane, meeting-pane, pane-registry, rdflib, solid-namespace, solid-ui, source-pane)
* [x] [mashlib](https://github.com/solid/mashlib/pull/93) (rdflib, solid-namespace, solid-panes, solid-ui)

**Comment by michielbdejong** [comment]
Hm, seems I broke something now - try:
```sh
npm install
npm run build
cd dist
npx serve
```
Then open http://localhost:5000 with your browser and open the developer console.
You'll see an error "panes is not defined". This is probably due to the build steps, not converting TS to JS when pushing to npm? Should hopefully not be too hard to fix, will debug.

**Comment by michielbdejong** [comment]
It seems to be due to https://github.com/solid/chat-pane/issues/40

**Comment by michielbdejong** [comment]
Should also include https://github.com/solid/solid-auth-client/pull/150

**Comment by michielbdejong** [comment]
## Step 2

Using the latest npm versions of:
* @babel/runtime@7.10.5
* @sinonjs/text-encoding@latest
* async@latest
* auth-header@1.0.0
* base64url@latest
* commander@6.0.0
* concat-stream@latest
* cross-fetch@latest
* fs-extra@latest
* isomorphic-fetch@2.2.1
* isomorphic-webcrypto@latest
* jsonld@3.1.1
* mime-types@latest
* node-fetch@latest
* n3@1.4.0
* standard-http-error@latest
* whatwg-url@latest
* xmldom@0.3.0

I'm updating npm dependencies for:
* [ ] @solid/jose (@sinonjs/text-encoding, base64url, isomorphic-webcrypto)
* [ ] @solid/oidc-rp (@solid/jose, base64url, isomorphic-webcrypto, node-fetch, standard-http-error, whatwg-url)
* [ ] solid-cli (@solid/oidc-rp)
* [ ] solid-rest (concat-stream, cross-fetch, fs-extra, mime-types)
* [ ] solid-auth-cli (@solid/cli, async, cross-fetch, jsonld, n3, solid-rest)
* [ ] solid-auth-client(@babel/runtime, auth-header, commander, isomorphic-fetch, @solid/oidc-rp)
* [ ] solid-auth-tls ()
* [ ] rdflib (jsonld, n3, solid-auth-cli, solid-auth-client, xmldom)
* [ ] pane-registry (rdflib)
* [ ] solid-ui (rdflib)
* [ ] chat-pane (solid-ui)
* [ ] contacts-pane (solid-ui)
* [ ] folder-pane (solid-ui)
* [ ] issue-pane (solid-ui)
* [ ] meeting-pane (solid-ui)
* [ ] source-pane (solid-ui)
* [ ] solid-panes (chat-pane, contacts-pane, folder-pane, issue-pane, meeting-pane, pane-registry, rdflib, solid-ui, source-pane)
* [ ] [mashlib (rdflib, solid-panes, solid-ui)

**Comment by michielbdejong** [comment]
Step 3 would be:
* [ ] solid-multi-rp-client (oidc-rp)
* [ ] solid-auth-manager (oidc-rp)
* [ ] node-solid-server (rdflib, solid-auth-manager, mashlib)

**Comment by michielbdejong** [comment]
Current status: I can compile solid-auth-client with @solid/oidc-rp@0.10.0 from npm but not when I link it using Lerna. Then `npm run build` just fails saying it can't find the imports. I'll try what happens when I install it as a git dependency.

**Comment by michielbdejong** [comment]
Full-stack build is successful in mashlib-dev now.
Tests are failing in node-solid-server though.
I think this can be worked around by not linking to local rdflib in NSS , but maybe I should just include step 3 now, so that it all works like it's supposed to in mashlib-dev, and we can all test properly.
I could also maybe set up the whole tree with git dependencies so it can be tested outside mashlib-dev.

**Comment by michielbdejong** [comment]
I ran `npm outdated` on all repos up to solid-auth-client, will continue from solid-auth-tls next week.

**Comment by michielbdejong** [comment]
Hm, https://github.com/solid/solid-auth-tls was archived and not forked anywhere. Will find out why solid-ui depends on it.

**Comment by michielbdejong** [comment]
![20200727_112339](https://user-images.githubusercontent.com/408412/88526084-c52ca600-cffb-11ea-9619-3b3ee4c26951.jpg)

**Comment by michielbdejong** [comment]
Also node-solid-ws, @solid/acl-check.

**Comment by michielbdejong** [comment]
Update complete and tests passing:
* [ ] [@solid/jose](https://github.com/solid/jose/pull/11) (@sinonjs/text-encoding, base64url, isomorphic-webcrypto)
* [ ] [@solid/oidc-rp](https://github.com/solid/oidc-rp/pull/34) (@solid/jose, base64url, isomorphic-webcrypto, node-fetch, standard-http-error, whatwg-url)
* [ ] [solid-cli](https://github.com/solid/solid-cli/pull/20) (@solid/oidc-rp)
* [ ] [solid-rest](https://github.com/jeff-zucker/solid-rest/pull/25) (concat-stream, cross-fetch, fs-extra, mime-types)
* [ ] [solid-auth-cli](https://github.com/jeff-zucker/solid-auth-cli/pull/15) (@solid/cli, async, cross-fetch, jsonld, n3, solid-rest)
* [ ] [solid-auth-client](https://github.com/solid/solid-auth-client/pull/163) (@babel/runtime, auth-header, commander, isomorphic-fetch, @solid/oidc-rp)
* [x] ~~solid-auth-tls ()~~
* [ ] [rdflib](https://github.com/linkeddata/rdflib.js/pull/425) (jsonld, n3, solid-auth-cli, solid-auth-client, xmldom)
* [ ] [pane-registry](https://github.com/solid/pane-registry/pull/8) (rdflib)
* [ ] [solid-ui](https://github.com/solid/solid-ui/pull/325) (pane-registry, rdflib, solid-auth-client, ~~solid-auth-tls,~~ solid-namespace)
* [ ] [chat-pane](https://github.com/solid/chat-pane/pull/43) (solid-ui)
* [ ] contacts-pane (solid-ui)
* [ ] [folder-pane](https://github.com/solid/folder-pane/pull/14) (solid-ui)
* [ ] [issue-pane](https://github.com/solid/issue-pane/pull/27) (solid-ui)
* [ ] [meeting-pane](https://github.com/solid/meeting-pane/pull/35) (solid-ui)
* [ ] [source-pane](https://github.com/solid/source-pane/pull/8) (solid-ui)
* [ ] [solid-panes](https://github.com/solid/solid-panes/pull/238) (chat-pane, contacts-pane, folder-pane, issue-pane, meeting-pane, pane-registry, rdflib, solid-ui, source-pane)
* [ ] [mashlib](https://github.com/solid/mashlib/pull/98) (rdflib, solid-panes, solid-ui)

**Comment by michielbdejong** [comment]
<img width="1280" alt="Screenshot 2020-07-28 12 55 10" src="https://user-images.githubusercontent.com/408412/88657139-bb757200-d0d1-11ea-81d3-12ee0b376cf6.png">

**Comment by michielbdejong** [comment]
Making progress! :) Now able to log in, but seeing these errors:
<img width="1280" alt="Screenshot 2020-07-29 11 56 48" src="https://user-images.githubusercontent.com/408412/88786405-c47a4800-d192-11ea-995c-9e145577f483.png">
<img width="1280" alt="Screenshot 2020-07-29 11 57 15" src="https://user-images.githubusercontent.com/408412/88786415-c6dca200-d192-11ea-92f0-a4a124f98098.png">

**Comment by michielbdejong** [comment]
Note you need to comment out
```
# webcrypto-shim.mjs
# webcrypto-liner.js
```
in the .gitignore of mashlib-dev/workspaces/isomorphic-webcrypto, otherwise the build of for instance mashlib will fail with:
```
ERROR in ../isomorphic-webcrypto/src/browser.mjs
Module not found: Error: Can't resolve './webcrypto-shim.mjs' in '/Users/michiel/gh/solid/mashlib-dev/workspaces/isomorphic-webcrypto/src'
 @ ../isomorphic-webcrypto/src/browser.mjs 1:0-30
 @ ../oidc-rp/src/AuthenticationResponse.js
 @ ../oidc-rp/src/RelyingParty.js
 @ ../oidc-rp/src/index.js
 @ ../solid-cli/src/SolidClient.js
 @ ../solid-auth-cli/src/index.js
 @ ../rdflib/esm/fetcher.js
 @ ../rdflib/esm/index.js
 @ ./src/index.ts
 @ multi ./src/index.ts

**Comment by michielbdejong** [comment]
I should look into https://github.com/interop-alliance/oidc-rp / https://github.com/anvilresearch/oidc-rp / https://github.com/solid/oidc-rp situation, thanks @SharonStrats for pointing me to https://github.com/solid/oidc-rp#deprecation-notice ! CC @dmitrizagidulin

**Comment by michielbdejong** [comment]
Also https://github.com/solid/keychain/pull/4

**Comment by michielbdejong** [comment]
Merged and published:
* [x] [@solid/jose](https://github.com/solid/jose/pull/11) (@sinonjs/text-encoding, base64url, isomorphic-webcrypto)
* [x] [@solid/oidc-rp](https://github.com/solid/oidc-rp/pull/34) (@solid/jose, base64url, isomorphic-webcrypto, node-fetch, standard-http-error, whatwg-url)
* [ ] [solid-auth-client](https://github.com/solid/solid-auth-client/pull/163) (@babel/runtime, auth-header, commander, isomorphic-fetch, @solid/oidc-rp)
* [ ] [rdflib](https://github.com/linkeddata/rdflib.js/pull/425) (jsonld, n3, solid-auth-cli, solid-auth-client, xmldom)
* [ ] [pane-registry](https://github.com/solid/pane-registry/pull/8) (rdflib)
* [ ] solid-namespace
* [ ] [solid-ui](https://github.com/solid/solid-ui/pull/325) (pane-registry, rdflib, solid-auth-client, ~~solid-auth-tls,~~ solid-namespace)
* [ ] [chat-pane](https://github.com/solid/chat-pane/pull/43) (solid-ui)
* [ ] contacts-pane (solid-ui)
* [ ] [folder-pane](https://github.com/solid/folder-pane/pull/14) (solid-ui)
* [ ] [issue-pane](https://github.com/solid/issue-pane/pull/27) (solid-ui)
* [ ] [meeting-pane](https://github.com/solid/meeting-pane/pull/35) (solid-ui)
* [ ] [source-pane](https://github.com/solid/source-pane/pull/8) (solid-ui)
* [ ] [solid-panes](https://github.com/solid/solid-panes/pull/238) (chat-pane, contacts-pane, folder-pane, issue-pane, meeting-pane, pane-registry, rdflib, solid-ui, source-pane)
* [ ] [mashlib](https://github.com/solid/mashlib/pull/98) (rdflib, solid-panes, solid-ui)
* [ ] [solid-multi-rp-client](https://github.com/solid/solid-multi-rp-client/pull/10) (oidc-rp)
* [x] [@solid/oidc-op](https://github.com/solid/oidc-op/pull/23) ()
* [x] [@solid/keychain](https://github.com/solid/keychain/pull/4)
* [ ] [oidc-auth-manager](https://github.com/solid/oidc-auth-manager/pull/51) (oidc-rp,  oidc-op, solid-multi-rp-client)
* [ ] [node-solid-ws](https://github.com/solid/node-solid-ws/pull/20) ()
* [ ] [@solid/acl-check](https://github.com/solid/acl-check/pull/34) (rdflib, solid-namespace)
* [ ] [node-solid-server](https://github.com/solid/node-solid-server/pull/1454) (rdflib, oidc-auth-manager, mashlib)

* [ ] [solid-cli](https://github.com/solid/solid-cli/pull/20) (@solid/oidc-rp)
* [ ] [solid-rest](https://github.com/jeff-zucker/solid-rest/pull/25) (concat-stream, cross-fetch, fs-extra, mime-types)
* [ ] [solid-auth-cli](https://github.com/jeff-zucker/solid-auth-cli/pull/15) (@solid/cli, async, cross-fetch, jsonld, n3, solid-rest)

**Comment by michielbdejong** [comment]
Will do the rest once I can publish https://www.npmjs.com/package/solid-auth-client.

@jeff-zucker do you want to tag and publish versions of solid-cli, solid-rest and solid-auth-cli? I started updating those because I thought I needed them in the dependency tree, but in the end I was able to drop them thanks to https://github.com/linkeddata/rdflib.js/pull/361. Btw, we should also do a dependency update for https://github.com/solid/data-kitchen, right? Do you want to work on that together?

**Comment by michielbdejong** [comment]
Done! :)

**Comment by jeff-zucker** [comment]
> Will do the rest once I can publish https://www.npmjs.com/package/solid-auth-client.
> 
> @jeff-zucker do you want to tag and publish versions of solid-cli, solid-rest and solid-auth-cli? I started updating those because I thought I needed them in the dependency tree, but in the end I was able to drop them thanks to [linkeddata/rdflib.js#361](https://github.com/linkeddata/rdflib.js/pull/361). Btw, we should also do a dependency update for https://github.com/solid/data-kitchen, right? Do you want to work on that together?

I am just now seeing this message ... my github notifications must be fubared ... absolutely I want to work together on this ... I have recently made progress with the newest electron and mashlib in Data Kitchen ... I have no problem with removing solid-auth-cli from the rdflib dependency tree BUT I would like to see a way to easily substitute a different fetcher so that solid-rest can be used to fetch non-http requests.

**Comment by michielbdejong** [comment]
CC @jaxoncreed ^
Do you want to coordinate this between the two of you? Let me know if I can help.

**Comment by jeff-zucker** [comment]
@jaxoncreed and @michielbdejong - I m not sure which  one of you I need to be talking with, could you clarify your respective roles in relation to rdflib/mashlib/SolidOS?  I'd like chat/zoom with one or both of you when you are available.

My issues in relation to Data Kitchen and other uses of rdflib/mashlib outside a standard browser context:  1) how to plan for transition from solid-auth-cli to solid-client-authn. 2) How to use an alternate fetch if default_fetcher is removed. 3) whether solid-rest should remain a dependency even when solid-cli and solid-auth-cli are removed.

@michielbdejong - I also would like to talk with you about using the crud tests (and other parts of the test suite) against file:// URIs - AFAIK solid-rest should be able to pass most or all of the tests other than ACL related things.

**Comment by michielbdejong** [comment]
My role is purely ad-hoc and volunteering, nobody is paying me to maintain these libraries.
I do use these libraries myself though (for instance, when I was writing Solid's Test Suite), and whenever I need to fix something for my own stuff, I also upstream my fix for others to benefit from.

I think Jackson is in mostly the same position.

If *you* have fixed something then if it works for you and doesn't break the tests, and it feels like the right change to make, then it's probably good to merge.

1) One thing I learned while working in the Solid stack is don't plan for future transition, always stick to what works now. :)

2) You can specify the fetcher when you instantiate rdflib. Check out how solid-logic does it: https://github.com/solid/solid-logic/blob/471cdc01babfa397b440aa38a2f658f9985c7c72/src/index.ts#L27
Another option is to specify window.solidFetcher as you can see here:
https://github.com/linkeddata/rdflib.js/commit/752b7407b5f0f9f39966dc3f458737f6ab1bef94#diff-eb275dd515bd68f45e5e14b12d3f4ff97dc9b1e52d742f0e66f56f67dccb21b0R755

3) up to you! :)

re crud tests against file:/// yeah, go for it! You can fork the solid-crud-tests repo and replace all the lines where is uses solid-auth-fetcher, like
https://github.com/solid/solid-crud-tests/blob/b9bb2ab/test/surface/create-container.test.ts#L3
Then just insert your file:///-aware library there and run `npm run jest`.

**Comment by jeff-zucker** [comment]
Thanks @michielbdejong, I've more or less figured these things out since posting, but it helps to have confirmation.  I also work adhoc as a volunteer so really appreciate all the work you both have put in. 

As it turns out, once we add global.solidFetcher in addition to window.solidFetcher (which isn't available in nodejs, see my PR #464) it is indeed easy to sub in an alternate fetcher in rdflib.  In terms of the future changes, I believe in supporting what works now while also keeping as open to what might change as possible.  I'm releasing a new solid-node-client that does this by giving the user a choice as to whether they want solid-auth-fetcher or solid-client-authn-node or something else as the auth fetcher.  I am hoping for a flexible plugin architecture that supports users bringing their own just about everything.  I am currently running into a snag with that in that if I import both solid-auth-fetcher and solid-client-authn-node in the same process, they clobber each other somehow.  Must dig!  But  I have it working that user can use either one, just not both.

As soon as I have time (yeah, right!) I'll dig into the crud-tests.

In terms of dependencies, I am now on to trying to figure out how to include data-kitchen in Solidos. Mashlib, panes, and solid-ui are so integral to data-kitchen that it makes sense to have it live connected inside SolidOS.  For now I am doing manual npm linking but it will be nice to eventually have it within the amazing generation framework you have built.  Data-kitchen is not as ready for prime time as the other parts of SolidOS, so politically, I am not sure when is the right time to include it.  Would value your thoughts.

**Comment by michielbdejong** [comment]
@jeff-zucker You can include data-kitchen at the end of https://github.com/solid/solidos/blob/master/scripts/release so that it gets built with the latest dependencies whenever someone runs that script. For that, it doesn't need to be read for prime time, as long as the build doesn't error.
